plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java

```diff
@@ -256,8 +257,12 @@ public HiveWriterFactory(
 
         requireNonNull(hiveSessionProperties, "hiveSessionProperties is null");
         this.sessionProperties = hiveSessionProperties.getSessionProperties().stream()
-                .collect(toImmutableMap(PropertyMetadata::getName,
-                        entry -> session.getProperty(entry.getName(), entry.getJavaType()).toString()));
+                .map(propertyMetadata -> immutableEntry(
+                        propertyMetadata.getName(),
+                        session.getProperty(propertyMetadata.getName(), propertyMetadata.getJavaType())))
+                // The session properties collected here are used for events only. Filter out nulls to avoid problems with downstream consumers
```

Comment:
Is there a reason why we do not care about properties which are set to `null` in events?

Reply:
I am not aware of any consumer of these events. This must be some extension point for someone.
The only usage i found looks like a dummy consumer, at least this is what the commit message says.
https://github.com/trinodb/trino/blob/cea58426e0a8f4cfe330bc805f88d12ab2d3fc4e/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveEventClient.java#L31-L38

Including `null` values could be a breaking change to downstream even consumers (e.g. if the `[Immutable]Map.copyOf(event.getSessionProperties())`. Converting values to `Optional` would be a breaking change to. In the absence of more information, i deemed that omitting null values is the least breaking of all options.